### ISSUE TITLE
jax.lax.psum_scatter on CPU ignores --xla_cpu_collective_timeout_seconds and fails after 30 s

### ISSUE THREAD

**Issue by tengyifei** [jax-ml/jax#40831]
### Description

On CPU with `jax.distributed`, `jax.lax.psum_scatter` (reduce-scatter) fails after 30 s when the processes arrive more than about 30 s apart.
It ignores `--xla_cpu_collective_timeout_seconds`, while `jax.lax.all_gather` waits as configured.

The attached `repro.py` (about 40 s per run; HLO dumps attached) starts two processes with one CPU device each and sets both XLA collective timeout flags to 3600 s.
Each process runs one warm-up collective, then process 1 waits 40 s before the second one.

```
python repro.py psum_scatter   # bug: exit code 1
python repro.py all_gather     # control: exit code 0
```

Expected: process 0 waits and both calls succeed, as with `all_gather` (process 0 succeeded after 40.0 s).
Observed on jax and jaxlib 0.12.0.dev20260920: `psum_scatter` failed in 3 of 3 runs, like this:

```
process 0: second psum_scatter FAILED after 30.0 s: UNKNOWN: Gloo ReduceScatter failed: [external/xla++third_party_ext+gloo/gloo/transport/tcp/buffer.cc:72] Read timeout [::1]:38624
process 1: second psum_scatter FAILED after 0.0 s: UNKNOWN: Gloo ReduceScatter failed: [external/xla++third_party_ext+gloo/gloo/transport/tcp/pair.cc:547] Connection closed by peer [::1]:2475
```

Suspected cause: `GlooCollectives::CreateCommunicators` (`gloo_collectives.cc`) never calls `gloo::Context::setTimeout()`, so the gloo context keeps gloo's default of 30 s.
The other collectives in `gloo_communicator.cc` pass XLA's timeout on each call, but `GlooCommunicator::ReduceScatter` ignores the timeout in its `Executor` argument.
gloo's `Buffer::waitRecv()` (`gloo/transport/tcp/buffer.cc`), which reduce-scatter uses, waits at most the timeout of a pair (gloo's connection between two ranks), and a pair copies the context's timeout when it is created.

Potential fix: call `setTimeout()` with XLA's collective timeout before `connectFullMesh()` in `CreateCommunicators`, or implement reduce-scatter with `UnboundBuffer` send and receive and per-call timeouts, as `CollectivePermute` and `AllToAll` do.

[repro.py](https://github.com/user-attachments/files/32453192/repro.py)
[module_0000.jit_collective.cpu_after_optimizations.txt](https://github.com/user-attachments/files/32453194/module_0000.jit_collective.cpu_after_optimizations.txt)
[module_0000.jit_collective.before_optimizations.txt](https://github.com/user-attachments/files/32453193/module_0000.jit_collective.before_optimizations.txt)

### System info (python version, jaxlib version, accelerator, etc.)

Versions: jax and jaxlib 0.12.0.dev20260920 and 0.11.1.dev20260810 (both fail), Python 3.13.15, Linux 6.12 x86-64, 64 CPUs.

**Comment by mattjj** [comment]
@penpornk

**Comment by hawkinsp** [comment]
I'm taking this one.
